src/main/java/jp/co/soramitsu/sora/crypto/Crypto.java

```diff
@@ -0,0 +1,145 @@
+package jp.co.soramitsu.sora.crypto;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.security.KeyPair;
+import java.security.PublicKey;
+import java.time.Instant;
+import java.util.LinkedList;
+import java.util.List;
+import javax.validation.constraints.NotNull;
+import jp.co.soramitsu.sora.crypto.algorithms.RawSignatureStrategy;
+import jp.co.soramitsu.sora.crypto.algorithms.RawSignatureStrategy.SignatureSuiteException;
+import jp.co.soramitsu.sora.crypto.algorithms.SignatureSuiteRegistry;
+import jp.co.soramitsu.sora.crypto.hash.RawDigestStrategy;
+import jp.co.soramitsu.sora.util.Bencoder;
+import lombok.Getter;
+import lombok.Setter;
+
+
+public class Crypto {
+
+  @Setter
+  @Getter
+  private RawDigestStrategy digestStrategy;
+
+  public Crypto(@NotNull RawDigestStrategy digestStrategy) {
+    this.digestStrategy = digestStrategy;
+  }
+
+  private byte[] createVerifyHash(VerifiableJson document, ProofProxy proof)
+      throws CreateVerifyHashException {
+    ByteArrayOutputStream os = new ByteArrayOutputStream();
+    Bencoder bencoder = new Bencoder(UTF_8, os);
+
+    // sanitize inputs
+    this.sanitizeDocument(document);
+    this.sanitizeProof(proof);
+
+    // encode input into a string
+    try {
+      bencoder.encode(document.serializeAsMap());
+      bencoder.encode(proof.serializeAsMap());
+    } catch (IOException e) {
+      throw new CreateVerifyHashException(e);
+    }
+
+    return digestStrategy.digest(os.toByteArray());
```

Comment:
I suggest to move os.toByteArray() into bencoder. At this line we'll call something like that bencoder.getEncodedBytes()